src/primval.rs

```diff
@@ -14,6 +14,8 @@ pub enum PrimVal {
 
     Ptr(Pointer),
     FnPtr(Pointer),
+    VtablePtr(Pointer, Pointer),
+    SlicePtr(Pointer, u64),
```

Comment:
The pair approach is still better - fat pointers are not primitive and this will only be more true in time. @solson can make a better arfumebt though. Also, shouldn't panic but rather return `Err`.